icu4c/source/i18n/reldtfmt.cpp

```diff
@@ -430,9 +430,9 @@ RelativeDateFormat::setContext(UDisplayContext value, UErrorCode& status)
         if ( fCapitalizationBrkIter == NULL && (value==UDISPCTX_CAPITALIZATION_FOR_BEGINNING_OF_SENTENCE ||
                 (value==UDISPCTX_CAPITALIZATION_FOR_UI_LIST_OR_MENU && fCapitalizationOfRelativeUnitsForUIListMenu) ||
                 (value==UDISPCTX_CAPITALIZATION_FOR_STANDALONE && fCapitalizationOfRelativeUnitsForStandAlone)) ) {
-            UErrorCode status = U_ZERO_ERROR;
-            fCapitalizationBrkIter = BreakIterator::createSentenceInstance(fLocale, status);
-            if (U_FAILURE(status)) {
+            UErrorCode status2 = U_ZERO_ERROR;
```

Comment:
Same issue with possible `U_MEMORY_ALLOCATION_ERROR` errors being silently ignore. I think it would be better to use the `status`  parameter from the caller.